src/application/application.cpp

```diff
@@ -226,12 +226,23 @@ void Application::Predict() {
                       config_.precise_float_parser);
     TextReader<int> result_reader(config_.output_result.c_str(), false);
     result_reader.ReadAllLines();
-    std::vector<std::vector<int>> pred_leaf(result_reader.Lines().size());
+
+    auto nrow = result_reader.Lines().size();
+    auto ncol = 0;
+    if (nrow > 0) {
+      ncol = result_reader.Lines()[0].size();
+    }
+    std::vector<int> pred_leaf;
+    pred_leaf.reserve(nrow * ncol);
+
     #pragma omp parallel for num_threads(OMP_NUM_THREADS()) schedule(static)
-    for (int i = 0; i < static_cast<int>(result_reader.Lines().size()); ++i) {
-      pred_leaf[i] = Common::StringToArray<int>(result_reader.Lines()[i], '\t');
+    for (size_t irow = 0; irow < nrow; ++irow) {
```

Comment:
The [compilation error in the CI](https://dev.azure.com/lightgbm-ci/lightgbm-ci/_build/results?buildId=16409&view=logs&j=e0c3ed1e-0c79-508f-2647-33e12d6ed3d4&t=5d0879b4-8100-51c6-0dab-aba0d8459adc&l=598) teaches me that `irow` must be a signed integer :smile:

Reply:
Thanks! I changed the relevant type in gbdt.cpp, too.